### PR TITLE
[ROCm] [Perf] Add AITER Flash Attention

### PR DESCRIPTION
**Pull request by tjtanaa**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose

Add AITER flash attention support to the diffusion attention.

It enables the use of aiter flash attention whenever possible.
The logic of the `is_aiter_found_and_supported` is imported from `vllm` main repo.

## Test Plan

Ensure accuracy of generated image is still close to the TORCH_SDPA output.

Ensure there are performance improvement using vbench benchmarking instruction.

## Test Result

Server Command

```bash
#! /bin/bash
export DIFFUSION_ATTENTION_BACKEND=FLASH_ATTN
# export DIFFUSION_ATTENTION_BACKEND=TORCH_SDPA
unset HF_HOME
export HF_HUB_HOME=/app/model

vllm serve Qwen/Qwen-Image --omni --port 8099
```

Benchmark Command:

```bash
python3 benchmarks/diffusion/diffusion_benchmark_serving.py \
	--base-url http://localhost:8099 \
	--model Qwen/Qwen-Image \
	--task t2i \
	--dataset vbench \
	--num-prompts 5
```

Performance on MI300X

| Metric | AITER Flash Attention (Run 1) |  AITER Flash Attention (Run 2) | Torch SDPA (Run 1) | Torch SDPA (Run 2) | Average Flash Attention | Average Torch SDPA | **Performance Gain** |
|--------|------------------------|------------------------|-------------------|-------------------|------------------------|-------------------|---------------------|
| **Benchmark Duration (s)** | 69.68 | 69.53 | 75.81 | 76.18 | 69.61 | 75.99 | **8.40% faster** |
| **Request Throughput (req/s)** | 0.07 | 0.07 | 0.07 | 0.07 | 0.07 | 0.07 | No change |
| **Latency Mean (s)** | 13.9351 | 13.9051 | 15.1612 | 15.2357 | 13.92 | 15.20 | **8.41% reduction** |
| **Latency Median (s)** | 13.9303 | 13.9000 | 15.1253 | 15.2818 | 13.92 | 15.20 | **8.45% reduction** |
| **Latency P99 (s)** | 14.0576 | 13.9704 | 15.3256 | 15.3420 | 14.01 | 15.33 | **8.61% reduction** |

Wan2.2

Baseline video

https://github.com/user-attachments/assets/1d51dcb9-1552-4899-9c3b-1a71fbee0f7b

With AITER FA backend

https://github.com/user-attachments/assets/fe03ebb6-c15a-4afb-aedb-a475d28b6a7d

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/diffusion/attention/backends/utils/fa.py [hunk 1]
```diff
@@ -15,6 +15,18 @@
...
 import torch.nn.functional as F
 
+from vllm_omni.utils.platform_utils import is_rocm
+
+if is_rocm():
+    from vllm._aiter_ops import is_aiter_found_and_supported
+
+    # Choose to enable this by default on ROCm
+    # Whenever possible as it is the fastest backend
+    if is_aiter_found_and_supported():
```
Comment (ZJY0516) [inline vllm_omni/diffusion/attention/backends/utils/fa.py:25]: what if `is_aiter_found_and_supported() == False`? May be raise a exception here.

File: vllm_omni/diffusion/attention/selector.py [hunk 2]
```diff
@@ -81,6 +81,21 @@ def get_attn_backend(head_size: int) -> type[AttentionBackend]:
...
+
+        # Choose to enable this by default on ROCm
+        # Whenever possible as it is the fastest backend
+        if is_aiter_found_and_supported():
+            if backend_name is None:
+                backend_name = "FLASH_ATTN"
+        else:
+            if backend_name == "FLASH_ATTN":
+                logger.warning(
+                    """Flash Attention requires GPU with compute capability >= 8.0 or < 10.0. "
```
Comment (ZJY0516) [inline vllm_omni/diffusion/attention/selector.py]: Does rocm also has compute capability?

Comment (tjtanaa) [reply vllm_omni/diffusion/attention/selector.py]: ROCM has compute capability, but it is rarely used because of its complicated nature

AMD Data Center GPU (Instinct GPU), GPU Arch are gfx9 : gfx90a (MI200), gfx942 (MI300), gfx950 (MI355).
AMD Consumer GPU (Radeon GPU), GPU Arch are gfx10, gfx11, gfx12: gfx 1151, gfx 1201. (They have compute capability larger than AMD DataCenter GPU.

compute capability of gfx1201 is (12, 0)
compute capability of gfx950 is (9, 5)

So in AMD ecosystem, we are using GPU arch instead of compute capability.

File: vllm_omni/diffusion/attention/backends/utils/fa.py (hunk 1 above)
Comment (tjtanaa) [reply vllm_omni/diffusion/attention/backends/utils/fa.py:25]: Done

File: vllm_omni/diffusion/attention/selector.py (hunk 2 above)
Comment (ZJY0516) [reply vllm_omni/diffusion/attention/selector.py]: So perhaps you should change the log here to align with the amd gpu

**Comment by tjtanaa** [comment]
@ZJY0516  I have addressed your comments.

File: vllm_omni/diffusion/attention/selector.py (hunk 2 above)
Comment (tjtanaa) [reply vllm_omni/diffusion/attention/selector.py]: Thank you for spotting the issue. I have updated the logs.

**Comment by tjtanaa** [comment]
Ready for another row of review

File: vllm_omni/diffusion/attention/selector.py [hunk 3]
```diff
@@ -81,6 +81,28 @@ def get_attn_backend(head_size: int) -> type[AttentionBackend]:
                                "Falling back to TORCH_SDPA backend."""
                 )
                 backend_name = "TORCH_SDPA"
+    elif detect_device_type() == "cuda" and is_rocm():
```
Comment (hsliuustc0106) [inline vllm_omni/diffusion/attention/selector.py:84]: @Gaohan123 @gcanlin please check whether we can import flash_attn npu backend in this way and remove the ascend_attn.py under backends folder

Comment (ZJY0516) [reply vllm_omni/diffusion/attention/selector.py:84]: I suggest we refactor this like npu when hardware plugin is ready

Comment (gcanlin) [reply vllm_omni/diffusion/attention/selector.py:84]: That could work. We also need to make the prefix of AttentionBackend hardware-agnostic from the semantic perspective.

Comment (gcanlin) [reply vllm_omni/diffusion/attention/selector.py:84]: This way is more like CustomOp. From the current attention, it's not indeed too complex like vLLM's AttentionBackend. If they wouldn't be more complex in the future, this way would be clearer.

**Comment by tjtanaa** [comment]
@ZJY0516 @hsliuustc0106 can we include this in the v0.14.0 milestone?

**Comment by tjtanaa** [comment]
@ZJY0516 @hsliuustc0106 ready for merging.
